app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt

```diff
@@ -216,45 +227,89 @@ class ProfileEditActivityTest {
       ProfileEditActivity.createProfileEditActivity(
         context = context,
         profileId = 1
       )
     ).use {
-      onView(withId(R.id.profile_delete_button)).perform(click())
-      onView(withText(R.string.profile_edit_delete_dialog_positive))
-        .inRoot(isDialog())
-        .perform(click())
       testCoroutineDispatchers.runCurrent()
-      if (context.resources.getBoolean(R.bool.isTablet)) {
-        intended(hasComponent(AdministratorControlsActivity::class.java.name))
-      } else {
-        intended(hasComponent(ProfileListActivity::class.java.name))
-      }
+
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_title)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_message)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_positive)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_negative)
     }
   }
 
   @Test
-  fun testProfileEdit_landscape_deleteProfile_checkReturnsProfileListOnTabletAdminControlOnPhone() {
+  fun testProfileEdit_deleteProfileSuccessful_dialogIsDisplayedLandscape() {
     launch<ProfileEditActivity>(
       ProfileEditActivity.createProfileEditActivity(
         context = context,
         profileId = 1
       )
     ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_delete_button)).perform(click())
       onView(isRoot()).perform(orientationLandscape())
-      onView(withId(R.id.profile_delete_button)).perform(scrollTo()).perform(click())
-      onView(withText(R.string.profile_edit_delete_dialog_positive))
-        .inRoot(isDialog())
-        .perform(click())
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_title)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_message)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_positive)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_negative)
+    }
+  }
+
+  @Test
+  fun testProfileEdit_deleteProfileSuccessful_clickCancel_closesDeleteDialog() {
+    launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
+        context = context,
+        profileId = 1
+      )
+    ).use {
       testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_title)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_message)
+      onView(withText(R.string.profile_edit_delete_dialog_negative)).perform(click())
+
+      activityTestRule.launchActivity(null)
+      testCoroutineDispatchers.advanceUntilIdle()
+
+      // Closing the dialog does not close the activity.
+      assertThat(activityTestRule.activity.isFinishing).isFalse()
```

Comment:
What does this have to do with verifying that the dialog isn't showing? We should be verifying that its text is no longer present once it's closed.

Reply:
Please check the update. Thank you